samza-kafka/src/main/scala/org/apache/samza/checkpoint/kafka/KafkaCheckpointManager.scala

```diff
@@ -187,92 +174,83 @@ class KafkaCheckpointManager(
+        if (envelopes.isEmpty || messages.isEmpty) {
+          // fully read the log
+          info("Got empty list of messages")
+          //done = true
+        }
+        else {
+          count += messages.size()
+          // check the key
+          for (msg: IncomingMessageEnvelope <- messages) {
+            val key = msg.getKey.asInstanceOf[Array[Byte]]
+            currentOffset = msg.getOffset().toLong
+            if (key == null)
+              throw new KafkaUtilException("While reading checkpoint stream encountered message without key.")
+
+            val checkpointKey = KafkaCheckpointLogKey.fromBytes(key)
+
+            if (!shouldHandleEntry(checkpointKey)) {
+              info("Skipping " + entryType + " entry with key " + checkpointKey)
             }
-
-            for (response <- fetchResponse.messageSet(checkpointTopic, 0)) {
-              offset = response.nextOffset
-              startingOffset = Some(offset) // For next time we call
-
-              if (!response.message.hasKey) {
-                throw new KafkaUtilException("Encountered message without key.")
-              }
-
-              val checkpointKey = KafkaCheckpointLogKey.fromBytes(Utils.readBytes(response.message.key))
-
-              if (!shouldHandleEntry(checkpointKey)) {
-                debug("Skipping " + entryType + " entry with key " + checkpointKey)
-              } else {
-                handleEntry(response.message.payload, checkpointKey)
-              }
+            else {
+              val bbuf: java.nio.ByteBuffer = ByteBuffer.wrap(msg.getMessage.asInstanceOf[Array[Byte]])
+              handleEntry(bbuf, checkpointKey)
             }
           }
-        } finally {
-          consumer.close()
-        }
-
-        loop.done
-        Unit
-      },
-
-      (exception, loop) => {
-        exception match {
-          case e: InvalidMessageException => throw new KafkaUtilException("Got InvalidMessageException from Kafka, which is unrecoverable, so fail the samza job", e)
-          case e: InvalidMessageSizeException => throw new KafkaUtilException("Got InvalidMessageSizeException from Kafka, which is unrecoverable, so fail the samza job", e)
-          case e: UnknownTopicOrPartitionException => throw new KafkaUtilException("Got UnknownTopicOrPartitionException from Kafka, which is unrecoverable, so fail the samza job", e)
-          case e: KafkaUtilException => throw e
-          case e: Exception =>
-            warn("While trying to read last %s entry for topic %s and partition 0: %s. Retrying." format(entryType, checkpointTopic, e))
-            debug("Exception detail:", e)
         }
       }
-    ).getOrElse(throw new SamzaException("Failed to get entries for " + entryType + " from topic " + checkpointTopic))
-
+    } finally {
+      systemConsumer.stop();
+    }
+    info("done reading %s msgs from checkpoint stream  %s:%s" format(count, systemName, checkpointTopic))
```

Comment:
s/done/Done, s/msg/messages
Re-write log line for consistency:
"Done reading 10 messages from checkpoint system: foo stream: bar"

Reply:
Done.